barretenberg/cpp/src/barretenberg/stdlib_circuit_builders/circuit_builder_base_impl.hpp

```diff
@@ -235,6 +235,7 @@ void CircuitBuilderBase<FF_>::add_recursive_proof(const AggregationObjectIndices
 {
     if (contains_recursive_proof) {
         failure("added recursive proof when one already exists");
+        ASSERT(0);
```

Comment:
maybe could do something else here. I think throw_or_abort doesn't work here for some reason (maybe wasm)